lib/src/main/scala/com/worksap/nlp/uzushio/lib/filters/MergeListTag.scala

```diff
@@ -0,0 +1,59 @@
+package com.worksap.nlp.uzushio.lib.filters
+
+import com.worksap.nlp.uzushio.lib.filters.base.DocFilter
+import com.worksap.nlp.uzushio.lib.cleaning.Document
+
+class MergeListTag extends DocFilter {
+  final val acceptedTags = Seq("li", "option")
+  final val mdListSymbol = "- "
+
+  def containsAcceptedTag(cssSelectorStrs: Seq[String]): Boolean = {
+    extractDescendantTag(cssSelectorStrs, acceptedTags) match {
+      case Some(_) => true
+      case None => false
+    }
+  }
+
+  def extractDescendantTag(
+      cssSelectorStrs: Seq[String],
+      tagNames: Seq[String]
+  ): Option[String] = {
+    val iter = cssSelectorStrs.reverse.iterator
+    var i = 0
+
+    while (iter.hasNext) {
+      val tagWithCSS = iter.next()
+      val tagWithAttrs = tagWithCSS.split("[#\\.]")
+      i += 1
+      if (acceptedTags.contains(tagWithAttrs.head)) {
+        return Option(tagWithCSS)
+      }
+    }
+    return None
+  }
+
+  override def checkDocument(doc: Document): Document = {
+    var paragraphs = doc.aliveParagraphs.to[Seq]
+
+    (0 until paragraphs.length - 1).foreach { i =>
+      val paragraph = paragraphs(i)
+      val nextParagraph = paragraphs(i + 1)
+      val isAccteptedTags = containsAcceptedTag(paragraph.cssSelectors) && containsAcceptedTag(
+        nextParagraph.cssSelectors
+      )
+
+      if (isAccteptedTags && paragraph.path == nextParagraph.path) {
+        val mergedParagraph = nextParagraph.copy(
+          text = List(paragraph.text, nextParagraph.text)
```

Comment:
This line will have O(n^2) complexity because strings are immutable in Java. Should not still cause much problems I think.

Reply:
If I can afford to change this line after completing various tasks, refactor it.

Reply:
I think we are OK with leaving it as is because
1) Lists should not be very large
2) Individual strings are not 1 character long